packages/prettier-plugin-java/test/unit-test/pattern-matching/_output.java

```diff
@@ -19,43 +19,92 @@ static String formatter(Object o) {
     return formatted;
   }
 
+  public boolean test(final Object obj) {
+    return (
+      obj instanceof final Integer x &&
+      (x == 5 || x == 6 || x == 7 || x == 8 || x == 9 || x == 10 || x == 11)
+    );
+  }
+
   void test(Buyer other) {
     return switch (other) {
       case null -> true;
-      case Buyer b && this.bestPrice > b.bestPrice -> true;
-      case Buyer b && this.bestPrice > b.bestPrice -> {
+      case Buyer b when this.bestPrice > b.bestPrice -> true;
+      case Buyer b when this.bestPrice > b.bestPrice -> {
         return true;
       }
-      case (Buyer b && this.bestPrice > b.bestPrice) -> true;
-      case Buyer b &&
+      case Buyer b when this.bestPrice > b.bestPrice -> true;
+      case Buyer b when
         this.bestPrice > b.bestPrice &&
         this.bestPrice > b.bestPrice &&
         this.bestPrice > b.bestPrice &&
-        this.bestPrice > b.bestPrice -> true;
-      case Buyer b &&
+        this.bestPrice > b.bestPrice
+      -> true;
+      case Buyer b when
```

Comment:
I'm wondering if it would be better if we add parenthesis on break ?

It would be something like that:

```java
void test(Buyer other) {
  return switch (other) {
    case null -> true;
    case Buyer b when this.bestPrice > b.bestPrice -> true;
    case Buyer b when this.bestPrice > b.bestPrice -> {
      return true;
    }
    case Buyer titi when (
      this.bestPriceaaaaaaaazzzzzaaaaaaaaaq > b.bestPrice
    ) -> true;
    case Buyer b when (
      this.bestPrice > b.bestPrice &&
      this.bestPrice > b.bestPrice &&
      this.bestPrice > b.bestPrice &&
      this.bestPrice > b.bestPrice
    ) -> true;
    case Buyer b when (
      this.bestPrice > b.bestPrice &&
      this.bestPrice > b.bestPrice &&
      this.bestPrice > b.bestPrice &&
      this.bestPrice > b.bestPrice
    ) -> {
      return true;
    }
    case Buyer toto when (
      this.bestPrice > b.bestPrice &&
      this.bestPrice > b.bestPrice &&
      this.bestPrice > b.bestPrice &&
      this.bestPrice > b.bestPrice
    ) -> true;
    case Buyer b when (
      this.bestPrice > b.bestPrice &&
      this.bestPrice > b.bestPrice &&
      this.bestPrice > b.bestPrice &&
      this.bestPrice > b.bestPrice
    ) -> {
      return true;
    }
    default -> false;
  };
}
```

I was trying to do it, but I still have one issue if the line is breaking just after the arrow :

```java
void test(Buyer other) {
  return switch (other) {
    case Buyer titi when (
      this.bestPriceaaaaaaaazzzzzaaaaaaaaaq > b.bestPrice
    ) -> true;
    case Buyer titi when this.bestPriceaaaaaazzzaaaaaaaaaq > b.bestPrice ->
    true;
  };
}
```

Reply:
I like this a lot better! It also helped simplify the printer logic in `switchRule`. Just pushed this improvement, let me know what you think.